lms/djangoapps/discussion_api/api.py

```diff
@@ -274,6 +277,7 @@ def get_thread_list(
         "last_activity_at".
     order_direction: The direction in which to sort the threads by. The only
         values are "asc" or "desc". The default is "desc".
+    discussion_id: Content-specific discussion topic ID
```

Comment:
At the data layer (i.e. comments service) each threads points to exactly one topic which is named `commentable_id`.  No distinction is made at that level between an "inline" topic (one that is associated with a specific block inside the course) and a "top-level" topic (a user-defined / user-assigned id and title that are stored as in a dict attribute of the course block itself).

Here's an example of where / how we make the distinction: https://github.com/edx/edx-platform/blob/master/lms/djangoapps/django_comment_client/utils.py#L395

IIRC the distinction is only relevant in the context of access rules (related to cohort config, see that same utils module above for details) or when we are deciding whether or not we should offer a link from this content to the relevant block in the course (i.e., if it's inline, offer a link).